File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.py
def solution(k, tangerine):
    answer = 0
    dic = {}
    for t in tangerine:
        if t not in dic:
            dic[t] = 1
        else:
            dic[t] += 1
            
    dic = sorted(dic.items(), key = lambda x:x[1], reverse= True)
    
    for key, value in dic:
        k -= value
        answer += 1
        if k <= 0:
            break
    return answer

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.java
import java.util.*;
class Solution {
    public int solution(int k, int[] tangerine) {
        int answer = 0;
        Map<Integer, Integer> hm = new HashMap<>();
        
        for(int t : tangerine){
            hm.put(t, hm.getOrDefault(t, 0) + 1);
        }
        
        List<Integer> keySet  = new ArrayList<>(hm.keySet());
        
        keySet.sort((o1, o2) -> hm.get(o2).compareTo(hm.get(o1)));
        
        
        for(int i = 0; i < keySet.size(); i++){
            k -= hm.get(keySet.get(i));
            answer += 1;
            
            if(k <= 0){
                break;
            }
        }
        return answer;
    }
}